### PR TITLE
WS Fixes

### PR DESCRIPTION
**Pull request by MrLotU**
Also needs https://github.com/apple/swift-nio/pull/744 and https://github.com/vapor/websocket/pull/31

### CODE REVIEW

File: Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift [hunk 1]
```diff
@@ -118,28 +117,32 @@ private final class HTTPClientUpgradeHandler<Upgrader>: ChannelInboundHandler wh
...
         // be out of the pipeline. Anything else is an error.
-        guard case .head(let res) = responsePart, res.status == .switchingProtocols, upgrader.isValidUpgradeResponse(res) else {
+        if case .head(let res) = responsePart, res.status == .switchingProtocols, upgrader.isValidUpgradeResponse(res) {
+            self.res = res
+        } else if case .end(_) = responsePart, let res = self.res {
+            upgrading = true
+            
+            removeExtraHandlers(ctx: ctx).then {
+                return self.upgrader.upgrade(ctx: ctx, upgradeResponse: res)
+                }.then { handler in
```
Comment (tanner0101) [inline Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: small formatting issue here.

File: Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift [hunk 2]
```diff
@@ -118,28 +117,32 @@ private final class HTTPClientUpgradeHandler<Upgrader>: ChannelInboundHandler wh
...
+                    let p: EventLoopPromise<Bool> = ctx.eventLoop.newPromise()
+                    let resultFuture = p.futureResult.map { _ -> Upgrader.UpgradeResult in
```
Comment (tanner0101) [inline Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: are we OK ignoring the `Bool` value here?

File: Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift (hunk 1 above)
Comment (MrLotU) [reply Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: This seems to be Xcode's preferred formatting, hitting CTRL + I (re-indent) changes nothing

File: Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift (hunk 2 above)
Comment (MrLotU) [reply Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: Yeah, the bool will only ever be `true`, as you can see in https://github.com/apple/swift-nio/blob/nio-1.12/Sources/NIO/ChannelPipeline.swift#L507

**Comment by MrLotU** [comment]
I'll create another PR on the websocket repo with a testcase :)

File: Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift (hunk 2 above)
Comment (MrLotU) [reply Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: To be sure, I asked the NIO guys: https://discordapp.com/channels/431917998102675485/448584561845338139/534780196516134912

Comment (MrLotU) [reply Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: PR up on the NIO side. Will be shipped in NIO 2 :)

File: Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift (hunk 1 above)
Comment (MrLotU) [reply Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift]: Fixed it 😄

**Comment by MrLotU** [comment]
@tanner0101 Bump :)

**Comment by penny-coin** [comment]
Hey @MrLotU, you just merged a pull request, have a coin! 

You now have 68 coins.

**Pull request merged**